### PR TITLE
[declare] Miscellaneous nits from my main dev tree

### PR DESCRIPTION
**Pull request by ejgallego**
Some small commits from my dev tree, I think these are for the most part quite trivial, tho we reduce the internal API a bit.

### CODE REVIEW

File: vernac/declare.mli [hunk 1]
```diff
@@ -144,17 +144,10 @@ val declare_variable
...
-  -> ?feedback_id:Stateid.t
-  -> ?section_vars:Id.Set.t
   -> ?types:types
   -> ?univs:Entries.universes_entry
-  -> ?eff:Evd.side_effects
-  -> ?univsbody:Univ.ContextSet.t
-  (** Universe-constraints attached to the body-only, used in
-     vio-delayed opaque constants and private poly universes *)
   -> constr
   -> Evd.side_effects proof_entry
```
Comment (ejgallego) [inline vernac/declare.mli:152]: Next PR will make this `unit proof_entry`

**Comment by ejgallego** [comment]
I've removed the dependency on this PR, as it seems it may take a bit to get throught.

**Comment by ejgallego** [comment]
Ping @coq/vernac-maintainers

**Review by SkySkimmer — APPROVED** [review]

**Comment by ejgallego** [comment]
Merci !

**Pull request merged**